### PR TITLE
add option to override chromedriver binary via environment variable

### PR DESCRIPTION
**Pull request by NyCodeGHG**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

The geckodriver package has an option to override the binary used for geckodriver with the GECKODRIVER_FILEPATH environment variable.
This PR adds the same option for chromedriver.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Polish (an improvement for an existing feature)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (improvements to the project's docs)
- [ ] Specification changes (updates to WebDriver command specifications)
- [ ] Internal updates (everything related to internal scripts, governance documentation and CI files)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Backport Request

[//]: # (The current `main` branch is the development branch for WebdriverIO v9. If your change should be released to the current major version of WebdriverIO (v8), please raise another PR with the same changes against the `v8` branch.)

- [ ] This change is solely for `v9` and doesn't need to be back-ported
- [x] Back-ported PR at `#12339`

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Comment by christian-bromann** [comment]
@NyCodeGHG I am not sure if we want to introduce an environment variable for this, you can enable this easily within your capabilities:

```ts
browserName: "chrome",
'wdio:chromedriverOptions': {
    binary: process.env.CHROMEDRIVER_FILEPATH
}
```

Would that work for you?

**Comment by NyCodeGHG** [comment]
@christian-bromann this PR is for upstreaming a patch we would otherwise apply ourselves in [nixpkgs](https://github.com/NixOS/nixpkgs). Because we want to ship webdriverio as a package, we can't do it this way, because that would require the user to edit their config.
The binaries downloaded by webdriverio do not work on NixOS, because they rely on the [FHS](https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard). Thats why we want to use the packaged versions instead.
We could roll this patch in our package, but I would prefer to upstream this.

File: packages/wdio-utils/src/node/startWebDriver.ts [hunk 1]
```diff
@@ -72,9 +72,11 @@ export async function startWebDriver (options: Options.WebDriver) {
...
 
         const { executablePath: chromeExecuteablePath, browserVersion } = await setupPuppeteerBrowser(cacheDir, caps)
-        const { executablePath: chromedriverExcecuteablePath } = chromedriverOptions.binary
-            ? { executablePath: chromedriverOptions.binary }
-            : await setupChromedriver(cacheDir, browserVersion)
+        const { executablePath: chromedriverExcecuteablePath } =
+            process.env.CHROMEDRIVER_FILEPATH ? { executablePath: process.env.CHROMEDRIVER_FILEPATH }
+                : chromedriverOptions.binary
+                    ? { executablePath: chromedriverOptions.binary }
+                    : await setupChromedriver(cacheDir, browserVersion)
```
Comment (christian-bromann) [inline packages/wdio-utils/src/node/startWebDriver.ts:79]:
```suggestion
        const { executablePath: chromedriverExcecuteablePath } =
            process.env.CHROMEDRIVER_FILEPATH
                ? { executablePath: process.env.CHROMEDRIVER_FILEPATH }
                : chromedriverOptions.binary
                    ? { executablePath: chromedriverOptions.binary }
                    : await setupChromedriver(cacheDir, browserVersion)
```

**Review by christian-bromann — COMMENTED** [review]
Makes sense! Can we add some docs about these environment variable here: https://github.com/webdriverio/webdriverio/blob/1fa445be69dd234804e96665ec51be19c7897637/website/docs/Capabilities.md#L86-L88 ?

**Comment by NyCodeGHG** [comment]
> Also, mind signing the CLA?

@christian-bromann My employers CLA Manager can't access the Corporate Dashboard. I'm not sure why.

**Comment by christian-bromann** [comment]
> My employers CLA Manager can't access the Corporate Dashboard. I'm not sure why.

Let me know if there is anything I can do to help or if I should just raise a copy of this PR myself.

**Comment by erwinheitzman** [comment]
Quick question, do you feel like we should add the same for the edgedriver?

**Comment by christian-bromann** [comment]
> Quick question, do you feel like we should add the same for the edgedriver?

I think so , but we should do it in the [`edgedriver`](https://github.com/webdriverio-community/node-edgedriver) package.

**Comment by christian-bromann** [comment]
@NyCodeGHG any updates on the CLA signing? Are you planning do sign it?

**Comment by NyCodeGHG** [comment]
Sorry for the delay, I can't sign it myself, still need to figure things out with my employer.

**Comment by christian-bromann** [comment]
@NyCodeGHG I went ahead and made the change myself. Please provide a CLA signature next time. Thank you for your contributions.

**Pull request closed without merging**